.github/workflows/build.yml

```diff
@@ -8,44 +8,87 @@ on:
   workflow_dispatch:
     inputs:
       logLevel:
-        description: 'Reason'     
+        description: 'Reason'
         required: false
         default: 'Update package'
 
 jobs:
   build:
     name: build
     runs-on: ubuntu-latest
-    permissions:
-      id-token: write
-      attestations: write
     strategy:
       matrix:
         target_arch: [x86_64, x86, arm64-v8a, armeabi-v7a]
       fail-fast: false
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
+      - name: Checkout repo
+        uses: actions/checkout@v4
         with:
           submodules: 'true'
 
-      - uses: nttld/setup-ndk@v1
+      - name: Setting up NDK
+        uses: nttld/setup-ndk@v1
         id: setup-ndk
         with:
-          ndk-version: r23c
+          ndk-version: r27c
           add-to-path: false
-      - run: ./build.sh ${{ matrix.target_arch }}
-        env:
-          NDK_TOOLCHAIN: ${{ steps.setup-ndk.outputs.ndk-path }}/toolchains/llvm/prebuilt/linux-x86_64
 
-      - name: Attest build provenance
-        uses: actions/attest-build-provenance@v1
-        with:
-          subject-name: dist-${{ matrix.target_arch }}
-          subject-path: dist
+      - name: Getting AOSP's source code
+        run: |
+          python3 ./get_source.py --tags platform-tools-35.0.2
+
```

Comment:
Why is this step necessary?

Reply:
In the build process done by lzhiyong yes. If you prefer use the submodules method and recreate a build process this will be unnecessary. I think this point will be resolved by the discussion in the main thread of comments.